src/Symfony/Component/FeatureFlag/FeatureRegistry.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\FeatureFlag;
+
+use Symfony\Component\FeatureFlag\Exception\FeatureNotFoundException;
+
+final class FeatureRegistry implements FeatureRegistryInterface
+{
+    /**
+     * @param array<string, (\Closure(): mixed)> $features
```

Comment:
Can we support both static and dynamic toggles?
```suggestion
     * @param array<string, bool|(\Closure(): mixed)> $features
```
Release toggles for instance, which are temporary and usually last no more than a week or two, generally have a fixed toggling decision. In those cases, there is no point in creating a Closure.
```php
$registry = new FeatureRegistry([
    'static_toggle' => true,
    'dynamic_toggle' => new DynamicToggle(),
]);
```

Reply:
Static feature is a real use-case, but the value can be anything `mixed`. Even a Closure could be used as feature value. Then we can't make the difference between the Closure that needs to be called and the one that needs to be returned as-this.

@yceruto I'd stick with the current implementation and require closure wrapping for static values.

```php
$registry = new FeatureRegistry([
    'static_toggle' => fn () => true,
    'dynamic_toggle' => new DynamicToggle(),
]);
```

@Jean-Beru In order to support invokable classes, the type must be `callable` :

```suggestion
     * @param array<string, (callable(): mixed)> $features
```

Reply:
well, `$invokable(...)` makes `Closure` viable (and preferred to me as it's a simple type)

Reply:
So it needs to be instantiated like this:
```php
$registry = new FeatureRegistry([
    'dynamic_toggle' => (new DynamicToggle())(...),
]);
```